src/hermite_multidimensional.hpp

```diff
@@ -0,0 +1,213 @@
+    ren_factor[0] = 1;
+
+    std::vector<int> nextPos(dim, 1);
+    std::vector<int> jumpFrom(dim, 1);
+    std::vector<int> ek(dim, 0);
+    std::vector<double> factors(resolution+1, 0);
+    int jump = 0;
+
+
+    for (ullint jj = 0; jj < Hdim-1; jj++) {
+
+        if (jump > 0) {
+            jumpFrom[jump] += 1;
+            jump = 0;
+        }
+
+
+        for (int ii = 0; ii < dim; ii++) {
+            std::vector<int> forwardStep(dim, 0);
+            forwardStep[ii] = 1;
+
+            if ( forwardStep[ii] + nextPos[ii] > resolution) {
+                nextPos[ii] = 1;
+                jumpFrom[ii] = 1;
+                jump = ii+1;
+            }
+            else {
+                jumpFrom[ii] = nextPos[ii];
+                nextPos[ii] = nextPos[ii] + 1;
+                break;
+            }
+        }
+
+        for (int ii = 0; ii < dim; ii++)
+            ek[ii] = nextPos[ii] - jumpFrom[ii];
+
+        int k = 0;
+        for(; k < static_cast<int>(ek.size()); k++) {
+            if(ek[k]) break;
+        }
+
+        ullint nextCoordinate = vec2index(nextPos, resolution);
+        ullint fromCoordinate = vec2index(jumpFrom, resolution);
+
+
+        for (int ii = 0; ii < dim; ii++) {
+            H[nextCoordinate] = H[nextCoordinate] + R(k, ii) * y(ii, 0);
+        }
+        H[nextCoordinate] = H[nextCoordinate] * H[fromCoordinate];
+
+        std::vector<int> tmpjump(dim, 0);
+
+        for (int ii = 0; ii < dim; ii++) {
+            if (jumpFrom[ii] > 1) {
+                std::vector<int> prevJump(dim, 0);
+                prevJump[ii] = 1;
+                std::transform(jumpFrom.begin(), jumpFrom.end(), prevJump.begin(), tmpjump.begin(), std::minus<int>());
+                ullint prevCoordinate = vec2index(tmpjump, resolution);
+                H[nextCoordinate] = H[nextCoordinate] - (static_cast<T>(jumpFrom[ii]-1))*static_cast<T>(R(k,ii))*H[prevCoordinate];
+
+            }
+        }
+
+    }
+
+    if (renorm) {
+        H = renormalization(H, dim, resolution);
+    }
+
+    return H;
+
+}
+
+
+
+/**
+ * A wrapper around the templated function hafnian::mode_elem. Returns photon number
+ * statistics of a Gaussian state for a given covariance matrix `mat`. Based in the
+ * MATLAB code available at: https://github.com/clementsw/gaussian-optics
+ *
+ * @param mat a flattened vector of size \f$2n^2\f$, representing an
+ *       \f$2n\times 2n\f$ row-ordered symmetric matrix.
+ * @param d a flattened vector of size \f$2n\f$, representing the first order moments.
+ * @param resolution highest number of photons to be resolved.
+ *
+ */
+std::vector<std::complex<double>> hermite_multidimensional_all(std::vector<double> &R_mat, std::vector<double> &y_mat, int &resolution, int &renorm) {
+    std::vector<std::complex<double>> R_matq(R_mat.begin(), R_mat.end());
+    std::vector<std::complex<double>> y_matq(y_mat.begin(), y_mat.end());
+
+    return hermite_multidimensional_cpp(R_matq, y_matq, resolution, renorm);
+}
```

Comment:
why is this wrapper needed? Since no quad precision is used here, can the above templated function simply be interfaced in `hafnian.pyx`?

Reply:
Removed the unnecessary wrapper.